osf/management/commands/sync_doi_metadata.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+import datetime
+import logging
+
+from django.core.management.base import BaseCommand
+from osf.models import Identifier
+from django.utils import timezone
+
+from framework.celery_tasks import app
+from framework.celery_tasks.handlers import enqueue_task
+
+logger = logging.getLogger(__name__)
+
+
+@app.task(name='osf.management.commands.sync_doi_metadata', max_retries=5, default_retry_delay=60)
+def sync_identifier_doi(identifier):
+    identifier.referent.request_identifier_update('doi')
+    identifier.modified = timezone.now()
+    identifier.save()
+    logger.info(f' doi update for {identifier.value} complete')
+
+
+def sync_doi_metadata(modified_date, batch_size=100, dry_run=True):
+    identifiers = Identifier.objects.filter(
+        category='doi',
+        deleted__isnull=True,
+        modified__lte=modified_date,
+    )[:batch_size]
+    logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
+                f'{identifiers.count()} identifiers to mint')
+
+    for identifier in identifiers:
+        if not dry_run:
+            enqueue_task(sync_identifier_doi.s(identifier))
+
+        logger.info(f'{"[DRY RUN]: " if dry_run else ""}'
+                    f' doi minting for {identifier.value} started')
+
+
+class Command(BaseCommand):
+    """ Adds updates all DOIs, will remove metadata for DOI bearing resources that have been withdrawn. """
+    def add_arguments(self, parser):
+        super().add_arguments(parser)
+        parser.add_argument(
+            '--dry_run',
+            action='store_true',
+            dest='dry_run',
+        )
+        parser.add_argument(
+            '--batch_size',
+            '-b',
+            type=int,
+            default=100,
+            help='number of dois to update in this batch.',
+        )
+        parser.add_argument(
+            '--modified_date',
```

Comment:
Should this be `required=True`? It's a required positional arg for `sync_doi_metadata` with no default.